instrumentation/lettuce-5.0/src/test/java/com/nr/lettuce5/instrumentation/Lettuce5InstrumentationTest.java

```diff
@@ -0,0 +1,170 @@
+import com.newrelic.agent.introspec.InstrumentationTestRunner;
+import com.newrelic.agent.introspec.Introspector;
+import com.nr.lettuce5.instrumentation.helper.Data;
+import com.nr.lettuce5.instrumentation.helper.RedisDataService;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+import reactor.core.publisher.Flux;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(InstrumentationTestRunner.class)
+@InstrumentationTestConfig(includePrefixes = { "io.lettuce.core", "io.lettuce.core.protocol", "io.netty.channel" })
+public class Lettuce5InstrumentationTest {
+
+    @Rule
+    public GenericContainer redis = new GenericContainer(DockerImageName.parse("redis:5.0.3-alpine"))
+            .withExposedPorts(6379);
+    private RedisDataService redisDataService;
+
+    @Before
+    public void before() {
+        redisDataService = new RedisDataService(redis);
+        redisDataService.init();
+    }
+
+    @Test
+    public void testSync() {
+        // given some data
+        String key = "syncKey";
+        String value = "syncValue";
+
+        // when sync 'set' called
+        String response = redisDataService.syncSet(key, value);
+
+        // then response should be key
+        assertEquals("Then response should be key", key, response);
+
+        // when 'get' called
+        String received = redisDataService.syncGet(key);
+
+        // then value returned
+        assertEquals("Get value", value, received);
+
+        // and 2 transactions have been sent
+        Introspector introspector = InstrumentationTestRunner.getIntrospector();
+        assertEquals("Finished transaction count", 2, introspector.getFinishedTransactionCount(1000));
+        Collection<String> transactionNames = introspector.getTransactionNames();
+        assertEquals("Transaction name count", 2, transactionNames.size());
+
+        String setTransactionName = "OtherTransaction/Custom/com.nr.lettuce5.instrumentation.helper.RedisDataService/syncSet";
+        String getTransactionName = "OtherTransaction/Custom/com.nr.lettuce5.instrumentation.helper.RedisDataService/syncGet";
+
+        // and transaction names are in collection
+        assertTrue("Should contain transaction name for 'set'", transactionNames.contains(setTransactionName));
+        assertTrue("Should contain transaction name for 'get'", transactionNames.contains(getTransactionName));
+
+        // and required datastore metrics are sent
+        DatastoreHelper helper = new DatastoreHelper(DatastoreVendor.Redis.name());
+        helper.assertAggregateMetrics();
+        // FIXME: Why is this 2 when 4.3 and 6 return 1?
```

Comment:
No idea why for 5, we seem to get double DB metrics.  6 and 4.3 with same test, only show 1